### ISSUE TITLE
Moquette server does not stop completely upon stopServer()

### ISSUE THREAD

**Issue by spetrov** [moquette-io/moquette#81]
Hi, 

I'm trying to use the moquette server within an application, and it looks like it doesn't stop completely when the "stopServer()" method is invoked. 
I'm using version 0.7...

The following is the example I'm trying: 

```
package org.test.moquette;
import java.io.IOException;
import org.eclipse.moquette.server.Server;
public class Main {

    public static void main(String[] args) {
        Server  broker = new Server();

        try {
            broker.startServer();
        } catch (IOException e) {
            e.printStackTrace();
        }
        // do something here...
        try {
            Thread.sleep(10000);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }

        broker.stopServer();
    }
}
```

When I run this I see the following in the logs:

```
Aug 10, 2015 10:28:36 AM org.eclipse.moquette.spi.impl.SimpleMessaging processInit
INFO: Starting without ACL definition
Aug 10, 2015 10:28:36 AM org.eclipse.moquette.server.netty.NettyAcceptor initFactory
INFO: Server binded host: 0.0.0.0, port: 1883
Aug 10, 2015 10:28:36 AM org.eclipse.moquette.server.netty.NettyAcceptor initFactory
INFO: Server binded host: 0.0.0.0, port: 8080
Aug 10, 2015 10:28:46 AM org.eclipse.moquette.server.Server stopServer
INFO: Server stopping...
Aug 10, 2015 10:28:46 AM org.eclipse.moquette.spi.impl.SimpleMessaging onEvent
INFO: onEvent processing messaging event from input ringbuffer org.eclipse.moquette.spi.impl.events.StopEvent@796a56e5
Aug 10, 2015 10:28:46 AM org.eclipse.moquette.server.netty.NettyAcceptor close
INFO: Msg read: 0, msg wrote: 0
Aug 10, 2015 10:28:46 AM org.eclipse.moquette.server.netty.NettyAcceptor close
INFO: Bytes read: 0, bytes wrote: 0
Aug 10, 2015 10:28:46 AM org.eclipse.moquette.server.Server stopServer
INFO: Server stopped
```

However, it looks like there are still some threads running, and the server doesn't seem to have been stopped completely (see attached screenshot). 

![moquette-server-threads-after-stop](https://cloud.githubusercontent.com/assets/1580960/9178554/991a098a-3f4c-11e5-8862-cb9da93d252b.png)

Am I doing something wrong, or is that a known issue? 
Can you suggest a workaround for it? 

Thanks, 
Stoyan

**Comment by andsel** [comment]
Hi Stoyan
was bug on missed shutdown of ProtocolProcessor's Ring Buffer (the ring buffer used to flow data back on the cable) fixed with fdbee3a21f049b40d196ecd5404f77fb958a6f5b now seems to work.

 Andrea

**Comment by spetrov** [comment]
Hi Andrea, 
This issue seems to have come back after some recent changes. 
I tried it on master from Sep 30 (latest commit: https://github.com/andsel/moquette/commit/54c68a61c071530930f898ac45f960ab4f4021bf).

Can you please take a look when you have chance? 

Thanks, 
Stoyan
